src/build.d

```diff
@@ -1230,7 +1230,10 @@ void processEnvironment()
     }
     if (env.getNumberedBool("ENABLE_LTO"))
     {
-        dflags ~= ["-flto=full"];
+        if (env["HOST_DMD_KIND"] == "dmd")
```

Comment:
Is ldmd considered a different kind? If yes, then LGTM

Reply:
Yes, `HOST_DMD_KIND` is set to either `dmd`, `ldc2` or `gdc` based on the actual compiler.

See https://github.com/dlang/dmd/blob/019e5c34c0301cc1ed7d584f84e3e94eb520a3fa/src/build.d#L1159-L1195

Reply:
ok lets merge

Reply:
Actually GDC doesn't want the full part so maybe another branch is required

Reply:
Fixed, it will now use `-flto` for GDC